app/grandchallenge/core/widgets.py

```diff
@@ -19,6 +20,48 @@ class Media:
         js = ("vendored/jsoneditor/jsoneditor.min.js",)
 
 
+class ColorEditorWidget(forms.TextInput):
+    """
+    Widget that uses the vendored jscolor for editing a color.
+
+    Specify the color format by adding a format keyword:
+    >>> ColorEditorWidget(format="hex")
+
+    See the jscolor documentation for more
+
+    """
+
+    template_name = "coloreditor/coloreditor_widget.html"
+
+    class Media:
+        js = (
+            "vendored/jscolor/jscolor.min.js",
+            "js/coloreditor_widget.js",
+        )
+
+    def __init__(self, attrs=None, format="auto"):
+        self.format = format
+        super().__init__(attrs)
+
+    def get_context(self, name, value, attrs=None):
+        context = {}
+        context.update(self.attrs.copy() or {})
+        context.update(attrs or {})
```

Comment:
Not to sure about dumping all of the attrs into the context, better to be explicit about the ones that are used IMO.

Reply:
I also wasn't too sure, the Django docs suggest it's common practice for widgets and colorfield did this originally. So I took the safe route. However, I agree with you and it seems weird from a Pythonic way. So let's be explicit.

Reply:
I've looked more seriously at how the contexts are scoped for widgets and how the superclasses handle these. I've made my changes thus.